### ISSUE TITLE
'update' function implementation

### ISSUE THREAD

**Issue by noah-kuo** [modin-project/modin#2093]
### Describe the problem

While working with @devin-petersohn on implementing a progress bar, it seemed like the implementation of df.update() or other in-place operations might want to use different functions or names. Currently the progress bar says computation is being performed by a "caller" function.

### Source code / logs

![image](https://user-images.githubusercontent.com/31913344/93410125-43176b80-f84d-11ea-9e78-88f7bdb08486.png)

**Comment by YarShev** [comment]
Hi @noah-kuo , thanks for reporting!
Looks like the progress bar says correctly because [this function](https://github.com/modin-project/modin/blob/master/modin/backends/pandas/query_compiler.py#L141) is being performed if we call df.update() or series.update(). You suppose that the progress bar should say computation is being performed by a "update" function?

**Comment by devin-petersohn** [comment]
Yes @YarShev, it would be better to show `update`, which is the pandas API that was called. We wrap the `pandas.DataFrame.update` in the `caller` function because it is inplace and Modin does not allow inplace updates.

I am not sure that there is a nice solution to this problem.

**Comment by YarShev** [comment]
@devin-petersohn , okay, I will take a look on this.

**Comment by YarShev** [comment]
We could do it this way ([link](https://github.com/YarShev/modin/commit/80138ab421bd906fd43eda35908abfc0a3d2b66a)). @devin-petersohn , what do you think?

**Comment by YarShev** [comment]
#2092

**Comment by devin-petersohn** [comment]
Left a comment on your commit @YarShev

**Comment by YarShev** [comment]
I have created the PR to simplify discussion. Current comments:
Devin: what about `func.__name__` instead?
Yaroslav: There are some cases when we pass `lambda` as a parameter `func` to the `copy_df_for_func`. `func.__name__` will be `lambda` in that case. We already have the such example for `series_update`. Perhaps, we will worth adding some docstring on parameter `func_name` in order to facilitate better understanding of other people what is `func_name` needed for. What do you think?
